test/jdk/com/sun/net/httpserver/ExchangeAttributeTest.java

```diff
@@ -71,7 +71,7 @@ public static void setup() {
     public void testExchangeAttributes() throws Exception {
         var handler = new AttribHandler();
         var server = HttpServer.create(new InetSocketAddress(LOOPBACK_ADDR,0), 10);
-        server.createContext("/", handler);
+        server.createContext("/", handler).getAttributes().put("attr", "context-val");
```

Comment:
So this tests the new behaviour by verifying that a context attribute is no longer visible from the exchange attributes. I believe we also need to test the opposite: an exchange attribute is not visible from the context. Also we want to verify that setting the new system property actually revert to the old behavior.